plugins/modelcatalogresolver/main_test.go

```
@@ -0,0 +1,22 @@
+	"testing"
+
+	"github.com/maximhq/bifrost/core/schemas"
+	"github.com/maximhq/bifrost/framework/modelcatalog"
+)
+
+func TestResolveProviderFromCatalogPrefersOpenAIForHumeIntegration(t *testing.T) {
+	catalog := modelcatalog.NewTestCatalog(nil)
+	catalog.UpsertLive(schemas.Azure, "azure-key", false, []string{"shared-model"})
+	catalog.UpsertLive(schemas.OpenAI, "openai-key", false, []string{"shared-model"})
+
+	ctx := schemas.NewBifrostContext(nil, schemas.NoDeadline)
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)